src/execution/expression_executor.cpp

```diff
@@ -113,16 +113,19 @@ Value ExpressionExecutor::EvaluateScalar(ClientContext &context, const Expressio
 
 	Vector result(expr.return_type);
 	executor.ExecuteExpression(result);
+	if (!allow_unfoldable && result.GetVectorType() != VectorType::CONSTANT_VECTOR) {
+		throw InvalidTypeException(
+		    "Prevent throwing an InternalException through a failed assertion, catch this in TryEvaluateScalar");
+	}
 
-	D_ASSERT(allow_unfoldable || result.GetVectorType() == VectorType::CONSTANT_VECTOR);
```

Comment:
This is supposed to be an assertion - the bug is in the NodeJS UDF not correctly setting the constant vector flag. I have fixed this in #5691.

Reply:
Alright, thanks, that was the other option I was considering.
I'll just change this PR to only add this test then